packages/framer-motion/src/projection/node/create-projection-node.ts

```diff
@@ -1972,6 +1974,7 @@ export function createProjectionNode<I>({
 
         clearSnapshot() {
             this.resumeFrom = this.snapshot = undefined
+            this.hasCheckedOptimisedAppear = false
```

Comment:
Why are we resetting this?

Reply:
Because the animations are only suspended we would need to recheck on each subsequent layout animation. I'm gonna take another look at this PR though and see about making a better cleanup function so we know when we can stop doing this.